src/internal/obj/factory.go

```diff
@@ -429,6 +429,16 @@ func NewClientFromURLAndSecret(url *ObjectStoreURL, reverse ...bool) (c Client,
 	case "wasb":
 		// In Azure, the first part of the path is the container name.
 		c, err = NewMicrosoftClientFromSecret(url.Bucket)
+	case "minio":
+		// TODO(brendon): This is kind of janky
+		// This is to get the PFS tests working without the filesystem object store
+		// The alternative was writing to environment variables in dockertestenv.
+		if !strings.HasSuffix(os.Args[0], ".test") {
```

Comment:
Yeah, this is terrifying but I guess there isn't really a better way supported by golang.

Reply:
The environment variables I'm referring to are to set the secret.  I'm not too familiar with how we do secrets for {Put,Get}FileURL.  But we need a way for PFS to figure out what secrets to use.  Maybe calling the scheme `test://` would be better?